backend/src/main/java/gov/cdc/usds/simplereport/config/AuthorizationConfiguration.java

```diff
@@ -47,19 +56,42 @@
   @Retention(RUNTIME)
   @Target(METHOD)
   @PreAuthorize(
-      "( @"
-          + AUTHORIZER_BEAN
-          + ".userHasSiteAdminRole() || "
+      "( "
+          + SPEL_IS_SITE_ADMIN
+          + " || "
           + "("
           + SPEL_HAS_PERMISSION
           + "MANAGE_USERS"
           + ")"
           + " && "
-          + "@"
-          + AUTHORIZER_BEAN
-          + ".userIsInSameOrg(#userId)"
+          + SPEL_IS_NOT_IN_SAME_ORG
           + ") )")
-  public @interface RequireGlobalAdminUserOrPermissionManageTargetUser {}
+  public @interface RequirePermissionManageTargetUser {}
+
+  /**
+   * Require the current user to to be one of the administrative users ("superusers") or have the
+   * {@link UserPermission#MANAGE_USERS} permission for the organization containing user with UUID
+   * {@code userId}; and require the user to not be the user they are operating on.
+   *
+   * <p>NOTE: any method with this annotation must have a parameter {@code userId}.
+   */
+  @Retention(RUNTIME)
+  @Target(METHOD)
+  @PreAuthorize(
+      "( "
```

Comment:
I'm usually not a fan of expression languages due to how tough they make composing complex expressions from simple ones, and this one is pretty hard to follow.

I don't think this is actionable in this PR, but maybe Spring has a way to make this type of expression composition more readable? I think a good first step would be replacing the `SPEL_HAS_PERMISSION` static string with a method that takes a UserPermission and returns a complete clause, but that may not be compatible with Java annotation processing.

Reply:
yeah it's not compatible because the expression has to be a constant :\

Reply:
Well, I guess we're stuck with it then. Maybe in a future PR these complex predicates could be converted to single method calls on the authorizer bean? I'm not sure if there's anything beyond that we could do.

Reply:
Agreed that moving the complex predicates into explicit code of our own is probably the medium or long-term solution.